### PR TITLE
Migrate packs

### PR DESCRIPTION
**Pull request by ksdfg**
To allow people to migrate all stickers from one pack to this one, so that it's easier for people to shift from other bots to neko chan as well as allowing people to kang entire packs at once

### CODE REVIEW

File: telebot/modules/stickers.py [hunk 1]
```diff
@@ -55,6 +56,26 @@ def get_sticker(update: Update, context: CallbackContext):
         update.effective_message.reply_text("Please reply to a sticker for me to upload its PNG.")
 
 
+def _get_pack_num_and_name(user: User, bot: Bot):
```
Comment (akhilnarang) [inline telebot/modules/stickers.py:59]: Random nitpicks, return type ka bhi type hints

Comment (ksdfg) [reply telebot/modules/stickers.py:59]: I do intend to properly comment and add type hints everywhere once I get this to a point where people actually feel like neko chan is a better alternative to notamemebot

File: telebot/modules/stickers.py [hunk 2]
```diff
@@ -289,14 +297,149 @@ def kang(update: Update, context: CallbackContext):
...
+        # this leads to an AttributeError being raised
+        except AttributeError:
+            pass
+
+        stickers = stickers[1:]  # because the first sticker is already in the new pack now
+        sticker_pack = context.bot.get_sticker_set(pack_name)
+
+    for sticker in stickers:
+        # download sticker
+        context.bot.get_file(sticker.file_id).download(f'{update.effective_user.id}_migrate_sticker.png')
```
Comment (akhilnarang) [inline telebot/modules/stickers.py:366]: Add in a `{uuid.uuid4()}` maybe, just in case

File: telebot/modules/stickers.py [hunk 3]
```diff
@@ -289,14 +297,149 @@ def kang(update: Update, context: CallbackContext):
...
+
+    # get the pack to migrate the stickers into
+    pack_num, pack_name = _get_pack_num_and_name(update.effective_user, context.bot)
+    appended_packs = [pack_name]  # list of packs the stickers were migrated into
+
+    try:
+        sticker_pack = context.bot.get_sticker_set(pack_name)
+    except BadRequest:
+        # download sticker
+        context.bot.get_file(stickers[0].file_id).download(f'{update.effective_user.id}_migrate_sticker.png')
```
Comment (akhilnarang) [inline telebot/modules/stickers.py:343]: same as below

**Comment by akhilnarang** [comment]
Does TG API allow checking md5/sha1/any similar checksums of the sticker file? Would be useful to prevent duplicates in pack

File: telebot/modules/stickers.py (hunk 3 above)
Comment (ksdfg) [reply telebot/modules/stickers.py:343]: in case of multiple migrates by the same user? that's actually a good idea....

**Comment by ksdfg** [comment]
@akhilnarang tbf, I can check for duplicate with the file_id of the sticker, but then I'll have to check in all of the user's packs, and that'll just slow down the entire process waay too much.... plus there's no way of telling which packs are already deleted ones like Alok's earlier one

**Comment by akhilnarang** [comment]
RIP

File: telebot/modules/stickers.py (hunk 1 above)
Comment (akhilnarang) [reply telebot/modules/stickers.py:59]: Cool

File: telebot/modules/stickers.py (hunk 3 above)
Comment (ksdfg) [reply telebot/modules/stickers.py]: done in 8e6e1e9

**Pull request merged**